src/Features/Core/Portable/StackTraceExplorer/DotnetStackFrameParser.cs

```diff
@@ -2,53 +2,27 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Diagnostics.CodeAnalysis;
-using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
 
 namespace Microsoft.CodeAnalysis.StackTraceExplorer
 {
     internal sealed class DotnetStackFrameParser : IStackFrameParser
     {
-        private const string StackTraceAtStart = "at ";
-        private const string StackTraceSymbolAndFileSplit = " in ";
-
         /// <summary>
-        /// Tries to parse a StackFrame following convention from Environment.StackTrace
-        /// https://docs.microsoft.com/en-us/dotnet/api/system.environment.stacktrace has
-        /// details on output format and expected strings
-        /// 
-        /// Example:
-        /// at ConsoleApp4.MyClass.M() in C:\repos\ConsoleApp4\ConsoleApp4\Program.cs:line 26
+        /// Uses <see cref="StackFrameParser"/> to parse a line if possible
         /// </summary>
         public bool TryParseLine(string line, [NotNullWhen(true)] out ParsedFrame? parsedFrame)
         {
             parsedFrame = null;
+            var tree = StackFrameParser.TryParse(line);
 
-            if (!line.Trim().StartsWith(StackTraceAtStart))
```

Comment:
can you do this without the TrimAllocation?  a simple mechanism woudl be a regex, or just a manual walk of the string.

Reply:
oh.  you removed this code.  :)